src/Npgsql/TypeMapping/GlobalTypeMapper.cs

```diff
@@ -135,6 +146,22 @@ internal NpgsqlDbType ToNpgsqlDbType(DbType dbType)
         internal DbType ToDbType(Type type)
             => _typeToDbType.TryGetValue(type, out var dbType) ? dbType : DbType.Object;
 
+        internal string? ToPgTypeName(NpgsqlDbType npgsqlDbType)
+        {
+            string? typeName;
+            var primitiveNpgsqlType = npgsqlDbType & ~NpgsqlDbType.Array;
+            if (_npgsqlDbTypeToPgTypeName.TryGetValue(primitiveNpgsqlType, out typeName))
+                if ((npgsqlDbType & NpgsqlDbType.Array) != 0)
```

Comment:
It would be better to just populate _npgsqlDbTypeToPgTypeName once with the array entries, rather than doing this each time on lookup.